crates/telio-network-monitors/src/local_interfaces.rs

```diff
@@ -0,0 +1,93 @@
+use ipnet::Ipv4Net;
+use std::net::{IpAddr, Ipv4Addr};
+
+/// Function type of a function returning interface addresses from the system
+pub type GetIfAddrs = fn() -> std::io::Result<Vec<if_addrs::Interface>>;
+
+/// Function returning interface addresses from the system
+pub fn system_get_if_addr() -> std::io::Result<Vec<if_addrs::Interface>> {
+    if_addrs::get_if_addrs()
+}
+
+/// Method that returns vector of Interfaces on the system
+/// Filtering out meshnet and loopback IP
+pub fn gather_local_interfaces(
+    get_if_addr: GetIfAddrs,
```

Comment:
I think this method doesn't need to receive this fn pointer as parameter anymore right?

Reply:
and you can use `if_addrs::get_if_addrs()` directly

Reply:
No we need that still for the `gather_local_interface_filtering` otherwise we can't mock it.

Reply:
you could by simply doing:
```
#[cfg(not(test))]
use if_addrs::get_if_addrs as GetIfAddrs;
#[cfg(test)]
use tests::get_if_addrs as GetIfAddrs;
```

Reply:
so you wouldn't need `GetIfAddrs` type or `system_get_if_addr` function.